relayer/relays/beacon/syncer/syncer.go

```diff
@@ -362,6 +362,42 @@ func (s *Syncer) GetHeaderUpdate(blockRoot common.Hash) (HeaderUpdate, error) {
 	return headerUpdate, nil
 }
 
+func (s *Syncer) GetBlockRange(lastBlockHash, secondLastBlockHash common.Hash) (uint64, uint64, error) {
+	lastBlock, err := s.Client.GetBeaconBlock(lastBlockHash)
+	if err != nil {
+		logrus.WithError(err).Error("unable to fetch last block using hash")
+
+		return 0, 0, err
```

Comment:
Earlier this year I refactored the relayer to use a new error reporting strategy, one that reduces the amount of logging boilerplate. The problem I saw was that each error handling block logs the error, and then returns the error. This ends up in the same error being logged repeatedly throughout the call chain.

So instead, most of the relayer codebase now uses [error wrapping](https://golangbyexample.com/wrapping-and-unwrapping-error-golang/) to provide context to errors:
```suggestion
		return 0, 0, fmt.Errorf("fetch last block: %w", err)
```

The other recommended thing was to avoid using the phrases "failed", "unable", "cannot" in error messages, as they end up being printed redundantly.

Then at the top of the call chain, the final wrapped error is logged once, printing a complete error message like `error: sync: determine block range: fetch last block: network connection timed out`

More docs about error wrapping:
* https://go.dev/blog/go1.13-errors

Reply:
Thanks for reminding me, this is a nifty pattern! Addressed in https://github.com/Snowfork/snowbridge/pull/653/commits/45db3ef5727c1fa56e7c5b0f12b387bce7f0cacd.